# Author information 
This file belongs to Czaigany Woodley, student ID: i6181735

# Introduction
South America is the continent with the least amount of countries. 
<br>
This dataset contains information about South American countries, namely Brazil, Colombia, Argentina, Peru, Venezuela, Chile, Ecuador, Bolivia, Paraguay, Uruguay, Guyana, and Suriname. Brazil is the largest country in South America, and will be the focus of my report. 

## Research question
Which vaccination played a larger role in improving the number of under-five deaths per 1000 population in Brazil? *Focus on results per year*.

### Subresearch questions
a) Is there a relationship between percentage Hepatitis B immunization coverage among 1-year-olds and the number of under-five deaths per 1000 population in Brazil?
<br>
b) Is there a relationship between percentage Polio immunization coverage among 1-year-olds and the number of under-five deaths per 1000 population in Brazil? 

# Data preparation

In [1]:
import altair as alt
import pandas as pd
from ipywidgets import interact
from ipywidgets import AppLayout
from ipywidgets import fixed
import ipywidgets as widgets

In [2]:
health_factors = pd.read_csv("https://raw.githubusercontent.com/NHameleers/dtz2025-datasets/master/CountryHealthFactors.csv")
health_factors.head(10)

Country  Year      Status  Life expectancy   Adult Mortality  \
0  Afghanistan  2015  Developing              65.0            263.0   
1  Afghanistan  2014  Developing              59.9            271.0   
2  Afghanistan  2013  Developing              59.9            268.0   
3  Afghanistan  2012  Developing              59.5            272.0   
4  Afghanistan  2011  Developing              59.2            275.0   
5  Afghanistan  2010  Developing              58.8            279.0   
6  Afghanistan  2009  Developing              58.6            281.0   
7  Afghanistan  2008  Developing              58.1            287.0   
8  Afghanistan  2007  Developing              57.5            295.0   
9  Afghanistan  2006  Developing              57.3            295.0   

   infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   ...  \
0             62     0.01               71.279624         65.0      1154  ...   
1             64     0.01               73.523582         62.0       492  ...   
2             66     0.01               73.219243         64.0       430  ...   
3             69     0.01               78.184215         67.0      2787  ...   
4             71     0.01                7.097109         68.0      3013  ...   
5             74     0.01               79.679367         66.0      1989  ...   
6             77     0.01               56.762217         63.0      2861  ...   
7             80     0.03               25.873925         64.0      1599  ...   
8             82     0.02               10.910156         63.0      1141  ...   
9             84     0.03               17.171518         64.0      1990  ...   

   Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  Population  \
0    6.0               8.16         65.0        0.1  584.259210  33736494.0   
1   58.0               8.18         62.0        0.1  612.696514    327582.0   
2   62.0               8.13         64.0        0.1  631.744976  31731688.0   
3   67.0               8.52         67.0        0.1  669.959000   3696958.0   
4   68.0               7.87         68.0        0.1   63.537231   2978599.0   
5   66.0               9.20         66.0        0.1  553.328940   2883167.0   
6   63.0               9.42         63.0        0.1  445.893298    284331.0   
7   64.0               8.33         64.0        0.1  373.361116   2729431.0   
8   63.0               6.73         63.0        0.1  369.835796  26616792.0   
9   58.0               7.43         58.0        0.1  272.563770   2589345.0   

    thinness  1-19 years   thinness 5-9 years  \
0                   17.2                 17.3   
1                   17.5                 17.5   
2                   17.7                 17.7   
3                   17.9                 18.0   
4                   18.2                 18.2   
5                   18.4                 18.4   
6                   18.6                 18.7   
7                   18.8                 18.9   
8                   19.0                 19.1   
9                   19.2                 19.3   

   Income composition of resources  Schooling  
0                            0.479       10.1  
1                            0.476       10.0  
2                            0.470        9.9  
3                            0.463        9.8  
4                            0.454        9.5  
5                            0.448        9.2  
6                            0.434        8.9  
7                            0.433        8.7  
8                            0.415        8.4  
9                            0.405        8.1  

[10 rows x 22 columns]

In [3]:
#checking which countries are found in this dataset 
health_factors['Country'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Australia',
       'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain', 'Bangladesh',
       'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin', 'Bhutan',
       'Bolivia (Plurinational State of)', 'Bosnia and Herzegovina',
       'Botswana', 'Brazil', 'Brunei Darussalam', 'Bulgaria',
       'Burkina Faso', 'Burundi', "Côte d'Ivoire", 'Cabo Verde',
       'Cambodia', 'Cameroon', 'Canada', 'Central African Republic',
       'Chad', 'Chile', 'China', 'Colombia', 'Comoros', 'Congo',
       'Cook Islands', 'Costa Rica', 'Croatia', 'Cuba', 'Cyprus',
       'Czechia', "Democratic People's Republic of Korea",
       'Democratic Republic of the Congo', 'Denmark', 'Djibouti',
       'Dominica', 'Dominican Republic', 'Ecuador', 'Egypt',
       'El Salvador', 'Equatorial Guinea', 'Eritrea', 'Estonia',
       'Ethiopia', 'Fiji', 'Finland', 'France', 'Gabon', 'Gambia',
       'Georgia', 'Germany'

In [4]:
# selecting columns relevant to my research question 
health_factors_needed_column = health_factors.drop(
    ['Life expectancy ','Adult Mortality','infant deaths',
     'Alcohol','percentage expenditure','Measles ',' BMI ',
     'Total expenditure','Diphtheria ',' HIV/AIDS','GDP','Population',' thinness  1-19 years',
     ' thinness 5-9 years','Income composition of resources','Schooling',], axis=1)

In [5]:
# selecting rows relevant to my research question 
health_factors_data_brazil = health_factors.loc[health_factors.Country == 'Brazil', ['Year', 'under-five deaths ', 'Hepatitis B', 'Polio']]

# Exploring data and cleaning of data

In [6]:
health_factors_data_brazil.head(15)

Year  under-five deaths   Hepatitis B  Polio
352  2015                  47         96.0   98.0
353  2014                  49         96.0   96.0
354  2013                  52         96.0   96.0
355  2012                  54         96.0   96.0
356  2011                  57         98.0   98.0
357  2010                  61         96.0   99.0
358  2009                  64         99.0   99.0
359  2008                  68         96.0   99.0
360  2007                  73         99.0   99.0
361  2006                  79         99.0   99.0
362  2005                  85         98.0   99.0
363  2004                  93         96.0   99.0
364  2003                 100         97.0   99.0
365  2002                 109         92.0   99.0
366  2001                 118         91.0   99.0

## Missing data
There is no missing data regarding the rows and columns that are relevant to my research question. I plotted a table with the data of Brazil to confirm that there is no missing data and/or outliers (*see table above*). 

# Data description and visualization

In [7]:
# descriptive statistics
health_factors_data_brazil.loc[:, ['under-five deaths ', 'Hepatitis B', 'Polio']].mean()

under-five deaths     77.2500
Hepatitis B           96.1875
Polio                 98.3125
dtype: float64

The mean under-five deaths in Brazil, between 2001 and 2015, is 77.25 deaths per 1000 population. 
Additionally, the mean percentage  immunization coverage among 1-year-olds of Hepatitis B is 96.2%, while that of Polio is 98.3%. 

In [8]:
health_factors_data_brazil.loc[:, ['Hepatitis B', 'Polio']].mode()

Hepatitis B  Polio
0         96.0   99.0

Between 2001 and 2015, majority of the calendar years in Brazil had a 96% Hepatitis B immunization coverage among 1-year-olds, and a 99% Polio immunization coverage among 1-year-olds. 

In [9]:
hepB_lineplot = alt.Chart(health_factors_data_brazil).mark_line().encode(
    y=alt.Y('Hepatitis B', scale=alt.Scale(domain=[90, 100])),
    x=alt.X('Year', scale=alt.Scale(domain=[2000, 2015])), 
    tooltip = ['Year', 'Hepatitis B'],
    ).properties(
    title="Percentage Hepatitis B immunization coverage among Brazilian 1-year-olds"
    )

polio_lineplot = alt.Chart(health_factors_data_brazil).mark_line().encode(
    y=alt.Y('Polio', scale=alt.Scale(domain=[95, 100])),
    x=alt.X('Year', scale=alt.Scale(domain=[2000, 2015])), 
    tooltip = ['Year', 'Polio'], 
    ).properties(
    title="Percentage Polio immunization coverage among Brazilian 1-year-olds"
    )

under_five_death_lineplot = alt.Chart(health_factors_data_brazil).mark_line().encode(
    y=alt.Y('under-five deaths '),
    x=alt.X('Year', scale=alt.Scale(domain=[2000, 2015])), 
    tooltip = ['Year', 'under-five deaths '],
    ).properties(
    title="Under-five deaths per 1000 population in Brazil"
    )

hepB_lineplot | polio_lineplot | under_five_death_lineplot

alt.HConcatChart(...)

The graphs shows that there is a difference in trend between the hepatitis B - and the polio immunization coverage. 
The largest decrease in Hepatitis B immunization coverage is observed between 2000 and 2001, and the largest increase between 2002 and 2003. The Hepatitis B immunization coverage remains stagnant after 2012. 

<br>

On the other hand, Polio immunization coverage is stagnant between 2000 and 2010, followed by a large decrease between 2010 and 2012, and it slightly increases again in 2014. 

In [10]:
health_factors_yearly_data = health_factors_data_brazil['Year'].unique()

In [11]:
# creating a dataframe with one filter
def select_country_data(Year, df):
  return df.loc[df['Year'] == Year, ]

In [12]:
def visualize_country_data(Year, df):
  yearly_data = select_country_data(Year, df)

In [15]:
def yearly_data(Year, df):
  test_plot = df.loc[(df['Year'] == Year), ]
  polio_per_year = alt.Chart(test_plot).mark_point().encode(
      x=alt.X('Hepatitis B'),
      y=alt.Y('under-five deaths '),
      tooltip=['Year', 'under-five deaths ', 'Polio'],
  )
  return polio_per_year
interact(yearly_data, Year = health_factors_yearly_data, df=fixed(health_factors)); 

interactive(children=(Dropdown(description='Year', options=(2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 20…

In [14]:
def yearly_data(Year, df):
  test_plot2 = df.loc[(df['Year'] == Year), ]
  hepB_per_year = alt.Chart(test_plot2).mark_point().encode(
      x=alt.X('Polio'),
      y=alt.Y('under-five deaths '),
      tooltip=['Year', 'under-five deaths ', 'Polio'],
  )
  return hepB_per_year
interact(yearly_data, Year = health_factors_yearly_data, df=fixed(health_factors));

interactive(children=(Dropdown(description='Year', options=(2015, 2014, 2013, 2012, 2011, 2010, 2009, 2008, 20…

# Discussion 
It is difficult to answer my subresearch questions, because I did not succeed on displaying the R-square value in my scatterplot, and therefore, my conclusion are based solely on assumptions by observing the different graphs. 
### Subresearch questions
a) Is there a relationship between percentage Hepatitis B immunization coverage among 1-year-olds and the number of under-five deaths per 1000 population in Brazil? 
Yes there is a relationship between the two, and this is most visible in the graph of year 2008. There, the data is clustered nicely together and there are not many outliers.  
<br>
b) Is there a relationship between percentage Polio immunization coverage among 1-year-olds and the number of under-five deaths per 1000 population in Brazil? 
Yes there is a relationship between the two, and this is most visible in the graph of year 2010. There, the data is clustered nicely together and there are not many outliers.

# Conclusion
As previously mentioned, it was difficult to answer this question, because I did not succeed in calculating an R-square for each scatterplot. However, if I compare the three linegraphs, I would assume that Polio immunization coverage played a larger role in under-five deaths compared to Hepatitis B immunization coverage. This is due to the fact that  while Polio coverage remained 99%, the under-five deaths decreased rapidly, and once a large decrease was observed in polio coverage (between 2010 - 2012), the under-five deaths continued decreasing, but at a slower rate. 